src/stackedhist.jl

```diff
@@ -0,0 +1,32 @@
+@userplot StackedHist
+
+@recipe function f(h::StackedHist)
+    arrays = reverse(h.args[1])
+    elt = eltype(arrays)
+    if elt <: Histogram 
+        bins = arrays[1].edges[1] # edges is a tuple
+        weights_ary = hcat([x.weights for x in arrays]...) #reverse so it stacks from top to bottom
+    elseif elt <: Array{<:Real} # if raw datas
+        # if no bins given, guess by fitting first data array
+        bins = pop!(plotattributes, :bins, fit(Histogram, arrays[1]).edges[1])
```

Comment:
I'd recommend using the histogram binning mechanism in Plots instead https://github.com/JuliaPlots/Plots.jl/blob/master/src/recipes.jl#L462-L834 it's much better and has taken considerable effort. And `bins` can also be a symbol referring to alternative binning methods. Well actually the best might be to to take the binning functionality from Plots (and there's also some here in StatsPlots) and PR it upstream to StatsBase.

Reply:
ah, I think it all comes down to
https://github.com/JuliaPlots/Plots.jl/blob/bece5a2620700c2d2d4955732a600c8222674c50/src/recipes.jl#L702